crates/wit-parser/src/resolve.rs

```diff
@@ -1745,3 +1754,183 @@ impl<'a> MergeMap<'a> {
+            exports: Default::default(),
+            includes: Default::default(),
+            include_names: Default::default(),
+            package: None,
+        };
+        let world2 = World {
+            name: "world2".to_string(),
+            docs: Default::default(),
+            imports: Default::default(),
+            exports: Default::default(),
+            includes: Default::default(),
+            include_names: Default::default(),
+            package: None,
+        };
+        worlds.alloc(world1);
+        worlds.alloc(world2);
+
+        let mut interfaces: Arena<Interface> = Arena::new();
+        let interface1 = Interface {
+            name: Some("interface1".to_string()),
+            docs: Default::default(),
+            types: Default::default(),
+            functions: Default::default(),
+            package: None,
+        };
+        let interface2 = Interface {
+            name: Some("interface2".to_string()),
+            docs: Default::default(),
+            types: Default::default(),
+            functions: Default::default(),
+            package: None,
+        };
+        interfaces.alloc(interface1);
+        interfaces.alloc(interface2);
+
+        let mut types: Arena<TypeDef> = Arena::new();
+        let type1 = TypeDef {
+            name: Some("type1".to_string()),
+            owner: TypeOwner::None,
+            kind: TypeDefKind::Type(Type::Bool),
+            docs: Default::default(),
+        };
+        let type2 = TypeDef {
+            name: Some("type2".to_string()),
+            owner: TypeOwner::None,
+            kind: TypeDefKind::Record(Record {
+                fields: Default::default(),
+            }),
+            docs: Default::default(),
+        };
+        types.alloc(type1);
+        types.alloc(type2);
+
+        let mut packages: Arena<Package> = Arena::new();
+        let package1 = Package {
+            name: PackageName {
+                namespace: "foo".into(),
+                name: "package1".into(),
+                version: None,
+            },
+            docs: Default::default(),
+            interfaces: Default::default(),
+            worlds: Default::default(),
+        };
+        let package2 = Package {
+            name: PackageName {
+                namespace: "foo".into(),
+                name: "package2".into(),
+                version: None,
+            },
+            docs: Default::default(),
+            interfaces: Default::default(),
+            worlds: Default::default(),
+        };
+        packages.alloc(package1);
+        packages.alloc(package2);
+
+        let resolve = Resolve {
+            packages,
+            interfaces,
+            types,
+            worlds,
+            package_names: Default::default(),
+        };
+        let s = serde_json::to_string(&resolve).unwrap();
+        let json_resolve = JSON_RESOLVE.replace(" ", "").replace("\n", "");
+        assert_eq!(s, json_resolve);
```

Comment:
@ydnar I've added a dummy `Resolve` to match the JSON output

Reply:
Thanks for the tests here! I'm a bit worried about the maintainability of this over time though as updating this test is going to be a bit of a pain. Alternatively though could the `tests/ui/*.wit` tests be all automatically serialized to JSON? Similar to how `*.result` files are generated for the `ui/parse-fail/*.wit` tests could the `*.json` tests be automatically generated for all the WIT successful tests? That way it'd be easy to see changes-at-a-glance and additionally auto-update test expectations when anything changes.

Reply:
Sounds good. I will work on that (expecting delayed commits from me as I am attending WasmCon this week)

Reply:
Just updated this PR to commit all JSON files corresponding to valid WIT files. The test runner is able to compare and emitted JSON from `resolve` to commited JSON files similar to how `.result` files work.